cmd/buildah/config.go

```diff
@@ -145,14 +146,22 @@ func updateConfig(builder *buildah.Builder, c *cli.Context) {
 		}
 	}
 	if c.IsSet("entrypoint") {
-		entrypointSpec := make([]string, 3)
-		entrypointSpec[0] = "/bin/sh"
-		entrypointSpec[1] = "-c"
-		entrypointSpec[2] = c.String("entrypoint")
-		if len(strings.TrimSpace(c.String("entrypoint"))) == 0 {
-			builder.SetEntrypoint(nil)
+    		var entrypointJson []string
+    		err := json.Unmarshal([]byte(c.String("entrypoint")), &entrypointJson)
+	
+		if err != nil {
+			// it wasn't a valid json array, fall back to string
+			entrypointSpec := make([]string, 3)
+			entrypointSpec[0] = "/bin/sh"
+			entrypointSpec[1] = "-c"
+			entrypointSpec[2] = c.String("entrypoint")
+			if len(strings.TrimSpace(c.String("entrypoint"))) == 0 {
+				builder.SetEntrypoint(nil)
+			} else {
+				builder.SetEntrypoint(entrypointSpec)
+			}
 		} else {
-			builder.SetEntrypoint(entrypointSpec)
+			builder.SetEntrypoint(entrypointJson)
```

Comment:
@pixdrift I think this is pretty darn close.  Like we chatted about in IRC, it doesn't handle 

```
buildah config --entrypoint ["/usr/bin/myscript.sh", "myinputdata"]
```

but it does handle this:

```
buildah config --entrypoint `["/usr/bin/myscript.sh", "myinputdata"]`
```
which @rhatdan and @nalind were good with, so leave that as is. 

The one thing that I think needs to be added is prepending "/bin/sh -c" to the entrypoint spec.  Maybe:
```
        if c.IsSet("entrypoint") {
                entrypointSpec:= []string{"/bin/sh", "-c"}

                var entrypointJson []string
                err := json.Unmarshal([]byte(c.String("entrypoint")), &entrypointJson)

                if err != nil {
                        // it wasn't a valid json array, fall back to string
                        entrypointSpec = append(entrypointSpec, c.String("entrypoint"))
                        if len(strings.TrimSpace(c.String("entrypoint"))) == 0 {
                                builder.SetEntrypoint(nil)
                        } else {
                                builder.SetEntrypoint(entrypointSpec)
                        }
                } else {
                        for _, node := range entrypointJson {
                                entrypointSpec = append(entrypointSpec, node)
                        }
                        builder.SetEntrypoint(entrypointSpec)
                }
                builder.SetCmd(nil)
        }
```

Reply:
@TomSweeneyRedHat, '/bin/sh -c' is exactly what I am trying to avoid when specifying an ENTRYPOINT using this method. If you look at the ticket #732, you will see in my example that `/bin/sh -c` isn't appended when the Dockerfile ENTRYPOINT is specified as an array.

Reply:
Really no /bin/sh -c in docker with an array?  How does it get run then?  Every time I've sent stuff to entrypoint and /bin/sh -c isn't in play it's fallen over.  I admittedly have kept it relatively simple, i think I've always just sent a string to Buildah or Docker.  Does Docker expect that the user will also specify the shell themselves if they send in an array?

Reply:
No requirement to specify the shell if supplying an array. The test to prove the behaviour is easy to replicate, as per #732, just change how ENTRYPOINT is declared in the Dockerfile and `docker build`, `docker inspect`.

Reply:
Thanks for the reference.  Seems weird that Docker adds /bin/sh -c for a string, but not for an array.